abci/example/dummyapp/app/dummyapp.go

```diff
@@ -0,0 +1,108 @@
+// Package dummy extends the kvstore.Application to include random data.
+// The random data is generated via the PreprocessTxs abci method.
+//
+// The name is inspired by:
+// https://github.com/lazyledger/lazyledger-prototype/blob/2aeca6f55ad389b9d68034a0a7038f80a8d2982e/app_dummy.go#L7
+package dummy
+
+import (
+	"bytes"
+	"math/rand"
+	"sort"
+	"time"
+
+	"github.com/lazyledger/lazyledger-core/abci/example/kvstore"
+	"github.com/lazyledger/lazyledger-core/abci/types"
+	tmproto "github.com/lazyledger/lazyledger-core/proto/tendermint/types"
+	"github.com/lazyledger/lazyledger-core/types/consts"
+)
+
+type Application struct {
+	*kvstore.Application
```

Comment:
out of curiosity, why is the kvstore `Application` used here instead of the `BaseApplication`? Is the added logic in the kvstore needed?

Reply:
Good question. Yeah, we could have used BaseApplication instead but then we also would have to implement DeliverTx etc (the dummyapp generates data for you provided with the right flags but it should still be possible to submit Txs - the simplest meaningful thing was to re-use the logic in kvstore - we can change this later to like a pseudo PayForMessage accepting app where the user can either submit a Tx or PayForMessage which includes a message).